Staircase.js

```diff
@@ -1,37 +1,61 @@
+/*
+TODO:
+    * Improve documentation?
+*/
 function Staircase(stairs) {
   this.stairs = {};
   for (var i in stairs) {
     this.stairs[i] = stairs[i];
-    this.stairs[i].firstVal = stairs[i].firstVal;
-    this.stairs[i].down = stairs[i].down;
-    this.stairs[i].up = stairs[i].up || 1;
-    this.stairs[i].factor = stairs[i].factor;
-    this.stairs[i].direction = stairs[i].direction;
-    this.stairs[i].limits = stairs[i].limits;
-    this.stairs[i].operation = stairs[i].operation;
-    this.stairs[i].wait = stairs[i].wait || false;
-    this.stairs[i].sameStairMax = stairs[i].sameStairMax;
     this.stairs[i].name = i;
-    this.stairs[i].val = stairs[i].val || [this.stairs[i].firstVal];
-    this.stairs[i].active = stairs[i].active || false;
-    this.stairs[i].limitReached = stairs[i].limitReached || false;
-    this.stairs[i].successiveGood = stairs[i].successiveGood || 0;
+    // Check minimum requirements: first value and stepSize
+    if (typeof stairs[i].firstVal=="undefined" && typeof stairs[i].limits != "object")
+        throw new Error("No firstVal specified for "+i);
+    else if(stairs[i].hasOwnProperty("firstVal"))
+        this.stairs[i].firstVal = stairs[i].firstVal;
+    else
+        this.stairs[i].firstVal = stairs[i].limits[0]+((stairs[i].limits[1]-stairs[i].limits[0])/2); // use the midpoint between limits if limits are specified and firstVal is not
```

Comment:
You're probably right! It's not strictly necessary to have a defined firstVal if it can be calculated from the limits - perhaps a warning rather than throwing an error? 

Throwing an error is a viable choice, of course, it just enforces a particular usage pattern on the end user (which might be a good thing).

Reply:
I was meaning that it may be more simple to require a value for `firstVal` without defaulting to anything. And throw an error if the `firstVal` is missing.

```js
if (typeof stairs[i].firstVal=="undefined") {
  throw new Error("No firstVal specified for "+i);
} else {
  this.stairs[i].firstVal = stairs[i].firstVal;
}
if (typeof stairs[i].stepSize=="undefined") {
  throw new Error("No stepSize specified for "+i);
} else {
  this.stairs[i].stepSize = stairs[i].stepSize;
}
```

Reply:
No problem with that behaviour; if you think it's more straightforward it gets my vote.